lib/room.js

```diff
@@ -529,6 +529,9 @@ function handleSignalingEvents(room, signaling) {
           participant._unsubscribeTracks();
         });
         room.emit(state, room, error);
+        room.localParticipant.tracks.forEach(publication => {
+          publication.unpublish();
```

Comment:
@PikaJoyce - I assume you are calling this `unpublish` for some side effect - of unregistering listener right? Can you clarify which event does this clear.

Reply:
Calling unpublish on all of the tracks cleans up all of the listeners that have been added. 

Since I check for the listeners on these events `enabled, disabled, started, stopped, message`, they all get cleared when the publications are unpublished on `disconnected`